python/ray/serve/autoscaling_policy.py

```diff
@@ -119,8 +124,15 @@ def __init__(self, config: AutoscalingConfig):
         self.decision_counter = 0
 
     def get_decision_num_replicas(
-        self, current_num_ongoing_requests: List[float], curr_target_num_replicas: int
+        self,
+        curr_target_num_replicas: int,
+        current_num_ongoing_requests: List[float],
+        current_handle_queued_queries: float,
     ) -> int:
+
+        if current_handle_queued_queries > 0 and len(current_num_ongoing_requests) == 0:
+            return max(1, curr_target_num_replicas)
+
         if len(current_num_ongoing_requests) == 0:
             return curr_target_num_replicas
```

Comment:
this does ensure proper scale down to 0 actually?

Reply:
scale down to 0 is ensured by following the logics in this function. When len(current_num_ongoing_requests) is 0, it means there is no replica